azurerm/internal/services/keyvault/key_vault_key_resource.go

```diff
@@ -102,9 +102,9 @@ func resourceKeyVaultKey() *schema.Resource {
 				ForceNew: true,
 				ValidateFunc: validation.StringInSlice([]string{
 					string(keyvault.P256),
+					string(keyvault.P256K),
 					string(keyvault.P384),
 					string(keyvault.P521),
```

Comment:
Can we add the old value as a hardcoded string, and translate it so that existing configurations continue to work? We can add a TODO to remove it in v3.0.

```suggestion
					string(keyvault.P521),
					"SECP256K1", // TODO: remove this in v3.0 as it was renamed to keyvault.P256K
```